substrate/frame/support/src/genesis_builder_helper.rs

```diff
@@ -27,7 +27,7 @@ use sp_runtime::format_runtime_string;
 /// [`sp_genesis_builder::GenesisBuilder::create_default_config`]
 pub fn create_default_config<GC>() -> sp_std::vec::Vec<u8>
 where
-	GC: BuildGenesisConfig + Default,
+	GC: BuildGenesisConfig + Default + serde::Serialize,
```

Comment:
I think this may not be needed as `BuildGenesisConfig` already implements serialize through `MaybeSerialize` marker trait:
https://github.com/paritytech/polkadot-sdk/blob/f2a750ee86e72c9ab677aaf588d0a33ee8446bef/substrate/primitives/runtime/src/traits.rs#L1134-L1140
https://github.com/paritytech/polkadot-sdk/blob/f2a750ee86e72c9ab677aaf588d0a33ee8446bef/substrate/frame/support/src/traits/hooks.rs#L472